nomad/job_endpoint_hook_consul_ce_test.go

```diff
@@ -48,6 +51,13 @@ func TestJobEndpointHook_ConsulCE(t *testing.T) {
 	test.Eq(t, "infra", job.TaskGroups[0].Services[1].Cluster)
 	test.Eq(t, "nondefault", job.TaskGroups[0].Tasks[0].Services[0].Cluster)
 
+	test.SliceContains(t, job.TaskGroups[0].Constraints,
+		&structs.Constraint{
+			LTarget: "${attr.consul.partition}",
```

Comment:
This tests the presence of the new constraint for the default Consul cluster, but I'm not seeing any test for a non-default Consul cluster `attr.consul.%s.partition`?

Reply:
Technically we can't have a non-default cluster in CE and having the per-cluster constraint in the first place is probably moot. But can't hurt to add a test here for it. There'll be a follow-up ENT PR as well.

Reply:
Actually, I'm going to back up a bit here and remove the conditional from CE, because it's never valid so there's no point in doing that. We'll only see that in ENT.

Reply:
Done in 908b445

Reply:
Looks good!  Will `consulPartitionConstraint()` have the same signature in both?  Doesn't it (maybe) need `cluster` in ENT?

Reply:
It does need the `cluster` field in ENT, so that's why I moved it into the CE file.

Reply:
OH!  I missed that alllll the logic ended up in the CE file.  Don't mind me :sweat_smile: